compiler/src/dmd/backend/cod1.d

```diff
@@ -1252,17 +1252,17 @@ void getlvalue(ref CodeBuilder cdb,code *pcs,elem *e,regm_t keepmsk)
                 }
                 if (!I16 && (ss = isscaledindex(e11)) != 0)
                 {   // (v * scale) + const
-                    cdisscaledindex(cdb, e11, &idxregs, keepmsk);
-                    reg = findreg(idxregs);
+                    cdisscaledindex(cdb, e11, &idxregs1, keepmsk);
+                    reg = findreg(idxregs1);
                     pcs.Irm = modregrm(0, 0, 4);
                     pcs.Isib = modregrm(ss, reg & 7, 5);
                     if (reg & 8)
                         pcs.Irex |= REX_X;
                 }
                 else
                 {
-                    scodelem(cdb, e11, &idxregs, keepmsk, true); // load index reg
-                    setaddrmode(pcs, idxregs);
+                    scodelem(cdb, e11, &idxregs1, keepmsk, true); // load index reg
+                    setaddrmode(pcs, idxregs1);
                 }
                 goto Lptr;
```

Comment:
Would it have been simpler to replace `goto Lptr;` with `return Lptr();`?

Didn't look to see whether there's more after this switch statement though.

Reply:
Much better idea, thanks